src/components/organisms/OpmWidget.tsx

```diff
@@ -1,102 +1,131 @@
-  ];
+  const getOpmHealth = () => {
+    const setSeverityForCategory = (category) => {
+      category.forEach((item) => {
+        item.severity = anomalyMap[item.data] ? SEVERITY.HIGH : "";
+      });
+      if (category.some((item) => item.severity === SEVERITY.HIGH)) {
+        setSeverity(SEVERITY.HIGH);
+      }
+    };
+    const anomalyMap = {};
+    OPMHealth.healthStatusList.forEach((item) => {
+      anomalyMap[item.property] = item.anomaly;
+    });
+    setSeverityForCategory(locale);
+    setSeverityForCategory(shipment);
+    setSeverityForCategory(channel);
+    setSeverityForCategory(payment);
+  };
 
-  const channel: GridData[] = [
-    { data: "DSK", severity: SEVERITY.HIGH },
-    { data: "MWB" },
-    { data: "IPH" },
-    { data: "AND" },
-    { data: "CSC" },
-    { data: "MPL" },
-    { data: "ZAP" },
-    { data: "INS" },
-  ];
+  const onGridCardClick = (e, d: GridData) => {
+    setData(d);
+    setOpenOverlay(true);
+    op.current?.toggle(e);
+  };
 
-  const payment: GridData[] = [
-    { data: "CC" },
-    { data: "GC" },
-    { data: "ETC" },
-    { data: "PPL" },
-    { data: "KLA" },
-    { data: "AFP" },
-    { data: "SEP" },
-    { data: "SEPT" },
-  ];
+  useEffect(() => {
+    getGroupedWidgetData();
+  }, []);
 
   useEffect(() => {
-    if (locale) {
-      if (locale.some((item) => item.severity === SEVERITY.HIGH)) {
-        setSeverity(SEVERITY.HIGH);
-        return;
-      }
-    }
-    if (shipment) {
-      if (shipment.some((item) => item.severity === SEVERITY.HIGH)) {
-        setSeverity(SEVERITY.HIGH);
-        return;
-      }
-    }
-    if (channel) {
-      if (channel.some((item) => item.severity === SEVERITY.HIGH)) {
-        setSeverity(SEVERITY.HIGH);
-        return;
-      }
-    }
-    if (payment) {
-      if (payment.some((item) => item.severity === SEVERITY.HIGH)) {
-        setSeverity(SEVERITY.HIGH);
-        return;
-      }
-    }
-  }, [locale, shipment, channel, payment]);
+    getOpmHealth();
+  }, [payment.length]);
+
+  setInterval(getOpmHealth, 60000);
 
   return (
     <div
       className={`focus-room-widget-wrapper px-4 pt-1 pb-4 grid-cols-2 gap-4 ${
         severity ? getSeverityStyles(severity).boxShadow : ""
       }`}
     >
-      <GridTable
+      <GridCards
         title={FOCUS_ROOM_TITLES.LOCALE}
         columns={2}
         data={locale}
         dataClassName="text-xs"
+        onClick={onGridCardClick}
       />
-      <GridTable
+      <GridCards
         title={FOCUS_ROOM_TITLES.SHIPMENT}
         columns={3}
         data={shipment}
         dataClassName="text-xs"
+        onClick={onGridCardClick}
       />
-      <GridTable
+      <GridCards
         title={FOCUS_ROOM_TITLES.CHANNEL}
         columns={3}
         data={channel}
         dataClassName="text-xs"
       />
-      <GridTable
+      <GridCards
         title={FOCUS_ROOM_TITLES.PAYMENT}
         columns={3}
         data={payment}
         dataClassName="text-xs"
```

Comment:
i can see a onclick functionality for locale and for shipment but not there for channel and payment is it missed out or is there any other reason

Reply:
It was missed. Added that